geode-core/src/main/java/org/apache/geode/distributed/internal/InternalDistributedSystem.java

```diff
@@ -1084,7 +1084,7 @@ public void run() {
       t.join(MAX_DISCONNECT_WAIT);
     } catch (InterruptedException e) {
       Thread.currentThread().interrupt();
-      logger.warn("Interrupted while processing disconnect listener",
+      logger.warn("Interrupted while processing disconnect listener thread id = " + t,
```

Comment:
This results in Thread.getName(). If you want "thread id" like the log statement says, then you'll want to change this to t.getId().

You should also change the logger statement to use log4j2 syntax:
```
logger.warn("Interrupted while processing disconnect listener thread id = {}", t, e);
```